main.tf

```
@@ -91,14 +91,18 @@ locals {
     namespace   = local.namespace
     environment = local.environment
     stage       = local.stage
-    attributes  = lower(replace(join(local.delimiter, local.attributes), local.regex_replace_chars, local.replacement))
+    attributes  = local.id_case == "none" ? replace(join(local.delimiter, local.attributes), local.regex_replace_chars, local.replacement) : lower(replace(join(local.delimiter, local.attributes), local.regex_replace_chars, local.replacement))
   }
 
   labels = [for l in local.label_order : local.id_context[l] if length(local.id_context[l]) > 0]
 
-  id_full = local.id_case == "lower" ? lower(join(local.delimiter, local.labels)) : (
-    local.id_case == "title" ? title(join(local.delimiter, local.labels)) :
-  upper(join(local.delimiter, local.labels)))
+  formatted_labels = [
+    for l in local.labels :
+    local.id_case == "none" ? l : (
+      local.id_case == "title" ? title(l) : (local.id_case == "upper" ? upper(l) : lower(l))
+  )]
+
+  id_full = join(local.delimiter, local.formatted_labels)
```

If we format inputs above as I suggest, then we do not need `formatted_labels`
```suggestion
  id_full = join(local.delimiter, local.labels)
```